### PR TITLE
Adding DQN

### PR DESCRIPTION
**Pull request by RPC2**

### CODE REVIEW

File: src/il_representations/scripts/dqn_train.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    print('Loading data...')
+    for idx in range(dataset_length):
+        if idx % 1000 == 0:
+            print(f'Loading {idx}/{dataset_length}...')
+        obs, next_obs, action, reward, done = dict_dataset['obs'][idx], \
+                                              dict_dataset['next_obs'][idx], \
+                                              dict_dataset['acts'][idx], \
+                                              dict_dataset['rews'][idx], \
+                                              dict_dataset['dones'][idx]
+        # obs = preprocess_obs(th.tensor(obs),
```
Comment (qxcv) [inline src/il_representations/scripts/dqn_train.py:126]: Why is/isn't this part necessary?

File: src/il_representations/scripts/dqn_train.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    model_saver = ModelSaver(trainer.policy,
+                             save_dir=os.path.join(out_dir, 'snapshots'),
+                             save_interval_batches=save_every_n_batches)
+    # trainer.learn(total_timesteps=1000000,
+    #               eval_log_path=os.path.join(out_dir, 'progress.csv'))
+
+
+    for epoch in range(nominal_num_epochs):
+        print(f'Training [{epoch}/{nominal_num_epochs}] epochs...')
+        policy, n_update, loss = trainer.train(n_update_per_epoch,
```
Comment (qxcv) [inline src/il_representations/scripts/dqn_train.py:156]: Maybe explain that this is just doing gradient updates on the network, and _not_ collecting new rollouts? I wasn't sure whether this was the case initially.

**Review by qxcv — CHANGES_REQUESTED** [review]
A few requests for clarifying comments, but otherwise looks good!

File: src/il_representations/scripts/dqn_train.py (hunk 1 above)
Comment (RPC2) [reply src/il_representations/scripts/dqn_train.py:126]: This is to perform image augmentation. I have modified this part and added some comments.

File: src/il_representations/scripts/dqn_train.py (hunk 2 above)
Comment (RPC2) [reply src/il_representations/scripts/dqn_train.py:156]: Yes it doesn't collect new rollouts. I've added some comments to it.

File: src/il_representations/scripts/policy_utils.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+from stable_baselines3.dqn import CnnPolicy
+
+from il_representations.algos.encoders import BaseEncoder
+from il_representations.utils import freeze_params, print_policy_info
+from il_representations.policy_interfacing import EncoderFeatureExtractor
+
+
+# TODO (Cynthia): This is not compatible with current changes on procgen
+# branch. Update load_encoder and make_policy functions when procgen gets
+# merged into master branch.
```
[the remark this answers is not in the document]
Comment (RPC2) [reply src/il_representations/scripts/policy_utils.py]: Yes, just updated.

File: src/il_representations/scripts/chain_configs.py [hunk 4]
```diff
@@ -0,0 +1,507 @@
+from il_representations.scripts.utils import StagesToRun
```
Comment (qxcv) [inline src/il_representations/scripts/chain_configs.py:1]: It looks like this file has reappeared somehow. It should be deleted, right?

File: src/il_representations/scripts/hp_tuning.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
+from il_representations.algos import augmenters, pair_constructors, encoders, losses, batch_extenders
```
Comment (qxcv) [inline src/il_representations/scripts/hp_tuning.py:1]: Similar with this file, I thought we deleted it on `cyn-expt`.

File: src/il_representations/test_support/configuration.py [hunk 6]
```diff
@@ -103,7 +108,7 @@
             'algo': tune.grid_search(['bc']),
             'freeze_encoder': tune.grid_search([False])
         },
-        'env_cfg': tune.grid_search([ENV_CFG_TEST_CONFIGS[0]]),
+        'env_cfg': tune.grid_search([ENV_CFG_TEST_CONFIGS[2]]),
```
Comment (qxcv) [inline src/il_representations/test_support/configuration.py:111]: Does it matter that this is using `ENV_TEST_CONFIGS[2]`? (i.e. will it break if we change 2 to 1 or 0?) If so, it might be a good idea to add a comment explaining why, and maybe a corresponding comment to the definition of `ENV_TEST_CONFIGS` as well.

**Review by qxcv — COMMENTED** [review]
I'm happy with the fixes from before, but worried there might be some spurious changes. In particular:

- Why are `scripts/hp_tuning.py` and `scripts/chain_configs.py` now in this PR? I thought they were deleted on the `cyn-expt` branch. This might have been a merge issue.
- Why do we need the demonstration pickle files for MAGICAL and Procgen? At the moment I think the tests are just using webdataset files; is it necessary to have the pickles as well?
